src/Listener/RelatedModelsListener.php

```diff
@@ -118,6 +118,12 @@ public function publishRelatedModels(?string $action = null, ?EntityInterface $e
      */
     protected function _findOptions(Association $association): array
     {
+        if ($association instanceof Association\BelongsToMany) {
+            return [
+                'keyField' => $association->getPrimaryKey(),
+            ];
+        }
+
         return [
             'keyField' => $association->getBindingKey(),
         ];
```

Comment:
Alternatively, the code could look like this:

```suggestion
        $keyField = $association->getBindingKey();
        if ($association instanceof Association\BelongsToMany) {
            $keyField' = $association->getPrimaryKey();
        }

        return [
            'keyField' =>$keyField,
        ];
```